studio/components/interfaces/Settings/Integrations/GithubIntegration/GitHubIntegrationConnectionForm.tsx

```diff
@@ -0,0 +1,266 @@
+        <Popover_Shadcn_ open={open} onOpenChange={setOpen} modal={false}>
+          <PopoverTrigger_Shadcn_ asChild name="branch-selector">
+            <Button
+              disabled={isUpdatingProdBranch}
+              type="default"
+              size="medium"
+              ref={comboBoxRef}
+              className={cn(
+                'justify-start w-64',
+                productionPreviewBranch?.git_branch === undefined ? 'text-light' : 'text'
+              )}
+              icon={
+                productionPreviewBranch?.git_branch && (
+                  <Shield className="w-4 h-4 text-warning" strokeWidth={1} />
+                )
+              }
+              loading={isUpdatingProdBranch || isLoadingBranches}
+              iconRight={
+                <span className="grow flex justify-end">
+                  <IconChevronDown className={''} />
+                </span>
+              }
+            >
+              {productionPreviewBranch?.git_branch || 'Select a branch'}
+            </Button>
+          </PopoverTrigger_Shadcn_>
+          <PopoverContent_Shadcn_
+            className="p-0"
+            side="bottom"
+            align="start"
+            style={{ width: comboBoxRef.current?.offsetWidth }}
+          >
+            <Command_Shadcn_>
+              <CommandInput_Shadcn_ placeholder="Find branch..." />
+              <CommandList_Shadcn_>
+                <CommandEmpty_Shadcn_>No branches found</CommandEmpty_Shadcn_>
+                <CommandGroup_Shadcn_>
+                  {githubBranches?.map((branch) => {
+                    const active = branch.name === productionPreviewBranch?.git_branch
+                    return (
+                      <CommandItem_Shadcn_
+                        key={branch.name}
+                        value={branch.name}
+                        className="cursor-pointer w-full flex items-center justify-between"
+                        onSelect={() => {
+                          setOpen(false)
+                          onUpdateProductionBranch(branch.name)
+                        }}
+                      >
+                        <div className="flex items-center gap-2">
+                          {active ? (
+                            <Shield className="w-4 h-4 text-warning" strokeWidth={1} />
+                          ) : (
+                            <GitBranch className="w-4 h-4" strokeWidth={1} />
+                          )}
+                          {branch.name}
+                        </div>
+                        {branch.name === productionPreviewBranch?.git_branch && <IconCheck />}
+                      </CommandItem_Shadcn_>
+                    )
+                  })}
+                </CommandGroup_Shadcn_>
+              </CommandList_Shadcn_>
+            </Command_Shadcn_>
+          </PopoverContent_Shadcn_>
+        </Popover_Shadcn_>
+      </div>
+
+      <Form_Shadcn_ {...form}>
```

Comment:
This whole file seems a bit off.

Reply:
Isn't this functionality covered elsewhere ?

Reply:
Also looks like something I did, maybe we moved it here.

Reply:
these were just shifted from IntegrationSettings